src/rapids_singlecell/tools/_umap.py

```diff
@@ -149,37 +154,61 @@ def umap(
     )  # 0 is not a valid value for rapids, unlike original umap
 
     n_obs = adata.shape[0]
-    n_neighbors = neigh_params["n_neighbors"]
-    if neigh_params.get("method") == "rapids":
-        knn_dist = neighbors["distances"].data.reshape(n_obs, n_neighbors)
-        knn_indices = neighbors["distances"].indices.reshape(n_obs, n_neighbors)
-        pre_knn = (knn_indices, knn_dist)
+    if parse_version(cuml.__version__) < parse_version("24.10"):
```

Comment:
so the import above works in lower versions, but you only use it in higher versions? Why?

Reply:
what do you mean with that?

Reply:
Add a comment explaining this please!